VEDA_OS01/transcripts.py

```diff
@@ -0,0 +1,268 @@
+"""
+Transcript handlers.
+"""
+import json
+import logging
+import uuid
+
+import boto
+import django.dispatch
+import requests
+from boto.s3.key import Key
+from pysrt import SubRipFile
+from requests.packages.urllib3.exceptions import InsecurePlatformWarning
+from rest_framework import status
+from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from control.veda_val import VALAPICall
+from VEDA_OS01 import utils
+from VEDA_OS01.models import (TranscriptPreferences, TranscriptProcessMetadata,
+                              TranscriptProvider, TranscriptStatus,
+                              VideoStatus)
+
+requests.packages.urllib3.disable_warnings(InsecurePlatformWarning)
+
+LOGGER = logging.getLogger(__name__)
+TRANSCRIPT_SJSON = 'sjson'
+CIELO24_TRANSCRIPT_COMPLETED = django.dispatch.Signal(providing_args=['job_id', 'lang_code', 'org', 'video_id'])
+CIELO24_GET_CAPTION_URL = 'https://api.cielo24.com/api/job/get_caption'
+CONFIG = utils.get_config()
+
+
+class TranscriptError(Exception):
+    """
+    An error occurred during fetching transcript from cielo24.
+    """
+    pass
+
+
+class TranscriptFetchError(TranscriptError):
+    """
+    An error occurred during fetching transcript from cielo24.
+    """
+    pass
+
+
+class TranscriptConversionError(TranscriptError):
+    """
+    An error occurred during srt to sjson conversion.
+    """
+    pass
+
+
+class TranscriptUploadError(TranscriptError):
+    """
+    An error occurred during sjson upload to s3.
+    """
+    pass
+
+
+class AllowValidTranscriptProvider(AllowAny):
+    """
+    Permission class to allow only valid transcript provider.
+    """
+    def has_permission(self, request, view):
+        """
+        Check if request is from valid transcript provider.
+        """
+        try:
+            return CONFIG['transcript_provider_request_token'] == int(view.kwargs['token'])
+        except (KeyError, ValueError):
+            return False
+
+
+class Cielo24CallbackHandlerView(APIView):
+    """
+    View to handler Cielo24 callback requests.
+    """
+    permission_classes = (AllowValidTranscriptProvider,)
+
+    def get(self, request, **kwargs):
+        """
+        Handle Cielo24 callback request.
+        """
+        attrs = ('job_id', 'lang_code', 'org', 'video_id')
+        if not all([attr in request.query_params for attr in attrs]):
+            return Response({}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
There should be warning level logs about the given and missing values.

also, responding 400 would trigger Cielo24 to re-try requesting to the same callback. Although, I can't think of any drawback of this.